cesium/channel.go

```diff
@@ -54,7 +67,42 @@ func (db *DB) RetrieveChannel(_ context.Context, key ChannelKey) (Channel, error
 	if vOk {
 		return vCh.Channel, nil
 	}
-	return Channel{}, core.ChannelNotFound
+	return Channel{}, core.NewErrChannelNotFound(key)
+}
+
+// RenameChannel implements DB.
+func (db *DB) RenameChannel(_ context.Context, key ChannelKey, newName string) error {
+	db.mu.Lock()
+	defer db.mu.Unlock()
+
+	udb, uok := db.unaryDBs[key]
+	if uok {
+		if udb.Channel.Name == newName {
+			return nil
+		}
+		udb.Channel.Name = newName
+		err := meta.Create(udb.FS, db.metaECD, udb.Channel)
+		if err != nil {
+			return err
+		}
+		db.unaryDBs[key] = udb
+		return nil
+	}
+	vdb, vok := db.virtualDBs[key]
+	if vok {
+		if vdb.Channel.Name == newName {
+			return nil
+		}
+		vdb.Channel.Name = newName
```

Comment:
This is a race condition. What if somebody is reading from the channel name while it is being renamed? Same case with `unary`. This doesn't mean we should resolve. But we should make a note that we're ok with a race in this scenario because we can guarantee it wont cause bugs

Reply:
I don't think it's an issue – changing the name does not affect anything in cesium.

Reply:
I know its not an issue. We should leave a comment that we know its a race condition and we're intentionally choosing to ignore it because its not an issue